From 3f2e1870b2a68627e65d15326da996c33d34b99e Mon Sep 17 00:00:00 2001
From: Jiri Moskovcak <jmoskovc@redhat.com>
Date: Thu, 20 Dec 2012 15:06:00 +0100
Subject: [PATCH 68/69] Make get_dirsize_find_largest_dir less talkative.
 related to rhbz#847291:

"it outputs 'is not a problem directory' three times right after
non-problem directory is created. It also outputs the message
six times again after each irritation (creating a new problem).
It is not stuck in a loop but it outputs the message too many times."

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
 src/lib/dirsize.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/lib/dirsize.c b/src/lib/dirsize.c
index 9f0b1b2..fa1059a 100644
--- a/src/lib/dirsize.c
+++ b/src/lib/dirsize.c
@@ -53,10 +53,19 @@ double get_dirsize(const char *pPath)
 
 static bool this_is_a_dd(const char *dirname)
 {
+    /* Prevent get_dirsize_find_largest_dir() from flooding log
+     * with "is not a problem directory" messages
+     * if there are stray dirs in /var/spool/abrt:
+     */
+    int sv_logmode = logmode;
+    logmode = 0;
+
     struct dump_dir *dd = dd_opendir(dirname,
                 /*flags:*/ DD_OPEN_READONLY | DD_FAIL_QUIETLY_ENOENT | DD_FAIL_QUIETLY_EACCES
     );
     dd_close(dd);
+
+    logmode = sv_logmode;
     return dd != NULL;
 }
 
-- 
1.8.0.1

